services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +43,38 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  if(xQueueSend(thermalMgrQueueHandle, event, (TickType_t) 10) != pdPASS) return ERR_CODE_INVALID_QUEUE_MSG;
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+
+  /* Get temp */
+  float temp;
+  readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temp);
+
+  if(temp >= LM75BD_DEFAULT_HYST_THRESH){
+    overTemperatureDetected();
+  } else {
+    safeOperatingConditions();
+  }
```

Comment:
Since `osHandlerLM75BD` is called from an ISR, there are 2 problems with this approach:
- I2C transfers are slow and ISRs must execute quickly (otherwise other interrupts can get delayed). So you can't read the temperature from inside this ISR.
- Internally, readTempLM75BD will attempt to take a mutex with a non-zero block time. This is problematic because you can't block from an ISR.

A potential solution is to send an event to the thermal mgr telling it that this interrupt occurred. And then the thermal mgr will take care of it next time it runs.

Reply:
ah yes, good point. I have added a new event type `THERMAL_MGR_EVENT_OS_HANDLE`, called `thermalMgrSendEvent(&event)` from the  `osHandlerLM75BD ` and exit right away. I modified the `thermalMgr` with some logic to handle the overtemp shutdown. Does this sound right?